relayer/src/chain/cosmos.rs

```diff
@@ -109,6 +113,103 @@ pub struct CosmosSdkChain {
 }
 
 impl CosmosSdkChain {
+    /// Does multiple RPC calls the full node, and ensures
+    /// reachability and some basic functions are available.
+    ///
+    /// Currently this method does the following checks:
+    ///     - that the node responds OK to `/health` RPC call;
+    ///     - the node has transaction indexing enabled;
+    ///     - the SDK version is supported.
+    ///
+    /// Emits a log warning in case anything is amiss.
+    /// Exits early if any health check fails, without doing any
+    /// further checks.
+    fn health_checkup(&self) {
+        async fn do_health_checkup(chain: &CosmosSdkChain) -> Result<(), Error> {
+            let chain_id = chain.id();
+            let grpc_address = chain.grpc_addr.to_string();
+            let rpc_address = chain.config.rpc_addr.to_string();
+
+            // Checkup on the self-reported health endpoint
+            chain
+                .rpc_client
+                .health()
+                .await
+                .map_err(|e| Kind::HealthCheckJsonRpc {
+                    chain_id: chain_id.clone(),
+                    address: rpc_address.clone(),
+                    endpoint: "/health".to_string(),
+                    cause: e,
+                })?;
+
+            // Checkup on transaction indexing
+            chain
+                .rpc_client
+                .tx_search(Query::default(), false, 1, 1, Order::Ascending)
```

Comment:
For some reason I am getting the same error when running it with gaiad v4.2.1 and v5.0.0:

```
Jul 05 15:21:43.987  WARN Hermes health check failed for endpoint /tx_search on the Json RPC interface of chain ibc-0:http://127.0.0.1:26657/; caused by: Internal error: 
parse error near Unknown (line 1 symbol 1 - line 1 symbol 1):
""
 (code: -32603)
Jul 05 15:21:43.987  WARN some Hermes features may not work in this mode! 
Jul 05 15:21:43.997  WARN Hermes health check failed for endpoint /tx_search on the Json RPC interface of chain ibc-1:http://127.0.0.1:26557/; caused by: Internal error: 
parse error near Unknown (line 1 symbol 1 - line 1 symbol 1):
""
 (code: -32603)
```

Reply:
Hmm, I'm not sure I understand. 

This is the error which Hermes is expected to produce if the full node does not have tx indexing enabled. If you were running a [patched gaiad](https://gist.github.com/adizere/2c8d53920b7fed8f28e6195405a18458) (regardless of v.4.2.1 or v.5.0.0) then this warning is correct behavior. Were you running patched gaiad?

Reply:
Tentative fix in https://github.com/informalsystems/ibc-rs/pull/1145/commits/b175b7415846c7a3c92d547d7a8c4b4b0db33ef6